### PR TITLE
Returning HTTP 200 if Course is Missing Submissions or Assignments

### PR DESCRIPTION
**Pull request by clintonb**
The templates will adjust accordingly to inform the user of the problem.

@dsjen

### CODE REVIEW

File: analytics_dashboard/courses/tests/test_presenters.py [hunk 1]
```diff
@@ -376,6 +377,33 @@ def test_assignments(self):
...
+            with mock.patch('analyticsclient.course.Course.problems', side_effect=NotFoundError):
+                # With no assignment type set, the method should return all assignment types.
+                assignments = self.presenter.assignments()
+                expected_assignments = self.factory.present_assignments(include_submissions=False)
+
+                self.assertListEqual(assignments, expected_assignments)
+                self.assertIsNone(self.presenter.last_updated)
+
+                # With an assignment type set, the presenter should return only the assignments of the specified type.
+                self.maxDiff = None
```
Comment (dsjen) [inline analytics_dashboard/courses/tests/test_presenters.py]: What does setting `maxDiff` to `None` do?

Comment (clintonb) [inline analytics_dashboard/courses/tests/test_presenters.py]: It displays the complete diff of any unequal `dicts` or `lists`.

Comment (dsjen) [inline analytics_dashboard/courses/tests/test_presenters.py]: I see. Is it now removed because it was just for testing?

Comment (clintonb) [inline analytics_dashboard/courses/tests/test_presenters.py]: Yes.

**Comment by dsjen** [comment]
LGTM!  Please wait for tests to pass before merging.

**Pull request merged**